sdk-installer/src/run.py

```diff
@@ -13,55 +13,130 @@
+class Conan(PackageManager):
+    def __init__(self, verbose_logging: bool):
+        self._verbose_logging = verbose_logging
 
-    Returns:
-        str: The required version.
-    """
-    sdk_version: str = "0.13.0"
-    requirements_path = os.path.join(
-        get_workspace_dir(), "app", "requirements-velocitas.txt"
-    )
-    if os.path.exists(requirements_path):
-        with open(requirements_path, encoding="utf-8") as requirements_file:
-            for line in requirements_file:
-                if line.startswith("velocitas_sdk") or line.startswith("velocitas-sdk"):
-                    sdk_version = line.split("==")[1].strip()
+    def is_package_installed(
+        self, package_name: str, package_version: Optional[str] = None
+    ) -> bool:
+        search_pattern = package_name
+        if package_version is not None:
+            search_pattern = f"{search_pattern}@{package_version}"
+
+        output = subprocess.check_output(
+            ["conan", "search", package_name], encoding="utf-8"
+        )
+        return output.find("Existing package recipes:") != -1
+
+    def get_required_package_version(self, package_name: str) -> str:
+        """Return the required version of the C++ SDK.
+
+        Returns:
+            str: The required version.
+        """
+        sdk_version: str = "0.4.1"
+        with open(
+            os.path.join(get_workspace_dir(), "conanfile.txt"), encoding="utf-8"
+        ) as conanfile:
+            for line in conanfile:
+                if line.startswith(f"{package_name}/"):
+                    sdk_version = line.split("/", maxsplit=1)[1].split("@")[0].strip()
+
+        return sdk_version
+
+    def install_local_package(self, path: str) -> None:
+        subprocess.check_call(
+            ["conan", "export", "."],
+            stdout=subprocess.DEVNULL if not self._verbose_logging else None,
+            cwd=path,
+        )
+
+
+class Pip(PackageManager):
+    def __init__(self, verbose_logging: bool):
+        self._verbose_logging = verbose_logging
+
+    def is_package_installed(
+        self, package_name: str, package_version: Optional[str] = None
+    ) -> bool:
+        output = subprocess.check_output(
+            ["pip", "list", package_name], encoding="utf-8"
+        )
 
-    return sdk_version
+        package_found = False
+
+        output_lines: List[str] = output.splitlines()
+        for line in output_lines:
+            package_data = line.strip().split(" ")
+            if package_data[0].startswith(package_name):
+                package_found = True
+                if package_version is not None:
+                    package_found = package_data[1] == package_version
+
+        return package_found
+
+    def get_required_package_version(self, package_name: str) -> str:
+        """Return the required version of the Python SDK.
+
+        Returns:
+            str: The required version.
+        """
+        sdk_version: str = "0.13.0"
```

Comment:
same as above, can we get this default version from somewhere else? If not we should at least make it a global variable

Reply:
Removed hardcoded version